datafusion/functions-aggregate/src/array_agg.rs

```diff
@@ -123,26 +120,22 @@ impl AggregateUDFImpl for ArrayAgg {
             )?));
         }
 
-        if acc_args.sort_exprs.is_empty() {
+        if acc_args.ordering_req.is_empty() {
             return Ok(Box::new(ArrayAggAccumulator::try_new(
                 &acc_args.input_types[0],
             )?));
         }
 
-        let ordering_req = limited_convert_logical_sort_exprs_to_physical_with_dfschema(
-            acc_args.sort_exprs,
-            acc_args.dfschema,
-        )?;
-
-        let ordering_dtypes = ordering_req
+        let ordering_dtypes = acc_args
+            .ordering_req
             .iter()
             .map(|e| e.expr.data_type(acc_args.schema))
             .collect::<Result<Vec<_>>>()?;
 
         OrderSensitiveArrayAggAccumulator::try_new(
             &acc_args.input_types[0],
             &ordering_dtypes,
-            ordering_req,
+            acc_args.ordering_req.to_vec(),
```

Comment:
I wonder if we still need to pass dtypes separately since we pass the full physical expression list for ordering?